client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
```

Comment:
The reason why I've not opted for such a solution is because unfortunately it's not as straightforward (as it usually is with domains). The `isSubdomain` function is very simple and dumb - it checks for 2-3 characters long tokens between dots and assumes those are TLD/SLDs. But there are many >3 character long SLDs (`.school.za`, `.csiro.au`, `.film.hu`, [etc.](https://en.wikipedia.org/wiki/Second-level_domain)) and there are many 2-3 character long domain names (cnn.com is an obvious one, az.pl is another example). The only reliable way is to have a hardcoded list of all the possible SLDs - something we can't afford on the frontend and doesn't seem worth a trip to the backend. Hence my proposal to just use the TLD part. It's not ideal, but I think given the situation is the optimal solution (it's better to stick to something we are more certain of than to confuse the user with guessing if we're dealing with a TLD/SLD/subdomain).